### PR TITLE
Entity pickup/drop sound components

### LINKED ISSUE #4454: Need pickup and drop sounds for stuff
Ideally for drops you'd use it for both landing and drop.
Pickup you'll probably need a new event for inventory / hands combined as they're both separate from each other.

### PR DESCRIPTION
**Pull request by Morb0**
## About the PR 
Closes #4454

**Screenshots**

https://user-images.githubusercontent.com/14136326/182259816-d98db8d7-6591-45da-b58f-e83595b6b910.mp4

**Changelog**

### CODE REVIEW

File: Content.Server/Sound/EmitSoundSystem.cs [hunk 1]
```diff
@@ -58,6 +59,8 @@ public override void Initialize()
             SubscribeLocalEvent<EmitSoundOnActivateComponent, ActivateInWorldEvent>(HandleEmitSoundOnActivateInWorld);
             SubscribeLocalEvent<EmitSoundOnTriggerComponent, TriggerEvent>(HandleEmitSoundOnTrigger);
             SubscribeLocalEvent<EmitSoundOnUIOpenComponent, AfterActivatableUIOpenEvent>(HandleEmitSoundOnUIOpen);
+            SubscribeLocalEvent<EmitSoundOnPickupComponent, GettingPickedUpAttemptEvent>(HandleEmitSoundOnPickup);
```
Comment (metalgearsloth) [inline Content.Server/Sound/EmitSoundSystem.cs:62]: This isn't the correct event to use, this is raised on actionblocker to check if pickup is possible and can be called at any time.

**Review by metalgearsloth — CHANGES_REQUESTED** [review]

File: Content.Server/Sound/EmitSoundSystem.cs (hunk 1 above)
Comment (Morb0) [reply Content.Server/Sound/EmitSoundSystem.cs]: Made a new one, on the same principle as drop

File: Content.Shared/Hands/EntitySystems/SharedHandsSystem.Pickup.cs [hunk 2]
```diff
@@ -141,6 +142,9 @@ public virtual void DoPickup(EntityUid uid, Hand hand, EntityUid entity, SharedH
 
         Dirty(hands);
 
+        var pickMsg = new PickedUpEvent(uid);
+        RaiseLocalEvent(entity, pickMsg, true);
+
         var didEquip = new DidEquipHandEvent(uid, entity, hand);
         RaiseLocalEvent(uid, didEquip, false);
```
Comment (metalgearsloth) [inline Content.Shared/Hands/EntitySystems/SharedHandsSystem.Pickup.cs:150]: Isn't this GotEquippedHandEvent?

Comment (Morb0) [reply Content.Shared/Hands/EntitySystems/SharedHandsSystem.Pickup.cs:150]: yeah... it looks like it.

**Comment by Morb0** [comment]
@metalgearsloth I think it's done now.

**Review by metalgearsloth — APPROVED** [review]

**Pull request merged**